tls/s2n_client_key_exchange.c

```diff
@@ -203,17 +209,21 @@ int s2n_ecdhe_client_key_send(struct s2n_connection *conn, struct s2n_blob *shar
 
 int s2n_rsa_client_key_send(struct s2n_connection *conn, struct s2n_blob *shared_key)
 {
-    uint8_t client_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN];
-    client_protocol_version[0] = conn->client_protocol_version / 10;
-    client_protocol_version[1] = conn->client_protocol_version % 10;
+    uint8_t client_version[S2N_TLS_PROTOCOL_VERSION_LEN];
+    uint8_t legacy_client_version = MIN(conn->client_protocol_version, S2N_TLS12);
```

Comment:
This MIN seems like a bad idea. If a client does show up with a weird "TLS1.3" client hello message, which isn't really a thing, it's better if the handshake ultimately fails or succeeds based on mutual agreement about what happened on the wire.  With the MIN we'd change it to 1.2.

Reply:
we have been using "min" in a couple of places https://github.com/awslabs/s2n/blob/master/tls/s2n_client_hello.c#L394 https://github.com/awslabs/s2n/blob/master/tls/s2n_client_hello.c#L187 - do you see similar issues? it seems like the worst case scenario of the "weird" tls 1.3 client may not be that bad - we fail the handshake, as a result the TLS 1.3 client not implementing it's client hello correctly and the server not picking a key exchange without forward secrecy.